blockchain/state/state_object.go

```diff
@@ -500,6 +500,23 @@ func (self *stateObject) Code(db Database) []byte {
 	return code
 }
 
+// CodeSize returns the size of the contract code associated with this object,
+// or zero if none. This method is an almost mirror of Code, but uses a cache
+// inside the database to avoid loading codes seen recently.
+func (self *stateObject) CodeSize(db Database) int {
+	if self.code != nil {
+		return len(self.code)
+	}
+	if bytes.Equal(self.CodeHash(), emptyCodeHash) {
+		return 0
+	}
```

Comment:
This is the core change.
Now it does not call `setError` when trying to find code with emptyCodeHash.